toolchains/BUILD

```diff
@@ -17,3 +17,19 @@ toolchain(
     toolchain = ":bash_rbe_ubuntu1604",
     toolchain_type = "@bazel_tools//tools/sh:toolchain_type",
 )
+
+# A duplicate of @buildbuddy_toolchain//:ubuntu_cc_toolchain, but with
+# correct cpp compiler constraint_value.
+toolchain(
+    name = "ubuntu_cc_toolchain",
+    exec_compatible_with = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
+    ],
+    target_compatible_with = [
+        "@platforms//os:linux",
+        "@platforms//cpu:x86_64",
+    ],
+    toolchain = "@buildbuddy_toolchain//:ubuntu_local_cc_toolchain",
+    toolchain_type = "@bazel_tools//tools/cpp:toolchain_type",
+)
```

Comment:
can we fix this upstream instead of changing it here?

Reply:
I intend to merge this into our repo temporarily to iterate faster.
Once we have a stable design up and running, I will be doing some cleanup in the upstream repo and include this fix in place.